airbyte-integrations/connectors/source-mysql/src/main/kotlin/io/airbyte/integrations/source/mysql/MySqlSourceDebeziumOperations.kt

```diff
@@ -243,34 +243,45 @@ class MySqlSourceDebeziumOperations(
         val file = Field("File", StringFieldType)
         val pos = Field("Position", LongFieldType)
         val gtids = Field("Executed_Gtid_Set", StringFieldType)
+        
         jdbcConnectionFactory.get().use { connection: Connection ->
             connection.createStatement().use { stmt: Statement ->
-                val sql = "SHOW MASTER STATUS"
-                stmt.executeQuery(sql).use { rs: ResultSet ->
-                    if (!rs.next()) throw ConfigErrorException("No results for query: $sql")
-                    val mySqlSourceCdcPosition =
-                        MySqlSourceCdcPosition(
-                            fileName = rs.getString(file.id)?.takeUnless { rs.wasNull() }
-                                    ?: throw ConfigErrorException(
-                                        "No value for ${file.id} in: $sql",
-                                    ),
-                            position = rs.getLong(pos.id).takeUnless { rs.wasNull() || it <= 0 }
-                                    ?: throw ConfigErrorException(
-                                        "No value for ${pos.id} in: $sql",
-                                    ),
-                        )
-                    if (rs.metaData.columnCount <= 4) {
-                        // This value exists only in MySQL 5.6.5 or later.
-                        return mySqlSourceCdcPosition to null
+                val queries = listOf("SHOW MASTER STATUS", "SHOW BINARY LOG STATUS")
+                for (sql in queries) {
+                    try {
+                        stmt.executeQuery(sql).use { rs: ResultSet ->
```

Comment:
should `SHOW MASTER STATUS` not work because it's no longer supported, it's `stmt.executeQuery(sql)` which will throw a `SQLException`. There's no need to also wrap the `.use { .. }` block in the try-catch which handles unsupported statements.

Reply:
agree, removed `.use`